tests/certification/bindings/rabbitmq/rabbitmq_test.go

```diff
@@ -0,0 +1,558 @@
+		Step(app.Run("standardApp", fmt.Sprintf(":%d", appPort), application)).
+		Step(sidecar.Run("standardSidecar",
+			embedded.WithAppProtocol(runtime.HTTPProtocol, appPort),
+			embedded.WithDaprGRPCPort(grpcPort),
+			embedded.WithDaprHTTPPort(httpPort),
+			embedded.WithComponentsPath("./components/standard"),
+			runtime.WithOutputBindings(
+				binding_loader.NewOutput("rabbitmq", func() bindings.OutputBinding {
+					return binding_rabbitmq.NewRabbitMQ(log)
+				}),
+			),
+			runtime.WithInputBindings(
+				binding_loader.NewInput("rabbitmq", func() bindings.InputBinding {
+					return binding_rabbitmq.NewRabbitMQ(log)
+				}),
+			))).
+		Step("send and wait", test).
+		Step("interrupt network", network.InterruptNetwork(30*time.Second, nil, nil, "5672")).
+		Run()
+}
+
+func TestRabbitMQExclusive(t *testing.T) {
+	log := logger.NewLogger("dapr-components")
+	messages := watcher.NewUnordered()
+
+	ports, _ := dapr_testing.GetFreePorts(3)
+	grpcPort := ports[0]
+	httpPort := ports[1]
+	appPort := ports[2]
+
+	test := func(ctx flow.Context) error {
+		client, err := daprClient.NewClientWithPort(fmt.Sprintf("%d", grpcPort))
+		require.NoError(t, err, "Could not initialize dapr client.")
+
+		// Declare the expected data.
+		msgs := make([]string, numOfMessages)
+
+		for i := 0; i < numOfMessages; i++ {
+			msgs[i] = fmt.Sprintf("exclusive-binding: Message %03d", i)
+		}
+
+		messages.ExpectStrings(msgs...)
+
+		metadata := make(map[string]string)
+
+		ctx.Log("Invoking binding!")
+		for _, msg := range msgs {
+			ctx.Logf("Sending: %q", msg)
+
+			req := &daprClient.InvokeBindingRequest{Name: "exclusive-binding", Operation: "create", Data: []byte(msg), Metadata: metadata}
+			err := client.InvokeOutputBinding(ctx, req)
+			require.Error(ctx, err, "error publishing message")
+		}
+		return nil
+	}
+
+	application := func(ctx flow.Context, s common.Service) (err error) {
+		// Setup the input binding endpoints.
+		err = multierr.Combine(err,
+			s.AddBindingInvocationHandler("exclusive-binding", func(_ context.Context, in *common.BindingEvent) ([]byte, error) {
+				messages.Observe(string(in.Data))
+				ctx.Logf("Got message: %s", string(in.Data))
+				return []byte("{}"), nil
+			}))
+		return err
+	}
+
+	flow.New(t, "rabbitmq certification").
+		// Run the application logic above.
+		Step(dockercompose.Run(clusterName, dockerComposeYAML)).
+		Step("wait for rabbitmq readiness",
+			retry.Do(time.Second, 30, amqpReady(rabbitMQURL))).
+		Step(app.Run("standardApp", fmt.Sprintf(":%d", appPort), application)).
+		Step(sidecar.Run("standardSidecar",
+			embedded.WithAppProtocol(runtime.HTTPProtocol, appPort),
+			embedded.WithDaprGRPCPort(grpcPort),
+			embedded.WithDaprHTTPPort(httpPort),
+			embedded.WithComponentsPath("./components/exclusive"),
+			runtime.WithInputBindings(
+				binding_loader.NewInput("rabbitmq", func() bindings.InputBinding {
+					return binding_rabbitmq.NewRabbitMQ(log)
+				}),
+			))).
+		Step("send and wait", test).
```

Comment:
```suggestion
        // TODO: The following test function will always fail as expected because the sidecar didn't initialize the component (expected). This should be updated to look for a much more specific error signature however by reading the sidecar's stderr.
		Step("send and wait", test).
```